### PR TITLE
[NEXUS-4562] Log auth failure at debug level

### PR DESCRIPTION
**Pull request by adreghiciu**

### CODE REVIEW

**Comment by cstamas** [comment]
woohoo, kill 'em all!

**Comment by nabcos** [comment]
+1

**Comment by velo** [comment]
+1

File: nexus/nexus-rest-api/pom.xml [hunk 1]
```diff
@@ -163,6 +163,10 @@
       <groupId>org.apache.shiro</groupId>
       <artifactId>shiro-web</artifactId>
     </dependency>
+    <dependency>
```
Comment (bdemers) [inline nexus/nexus-rest-api/pom.xml:166]: merge manually, this dep was just added to the pom

**Comment by bdemers** [comment]
+1

File: nexus/nexus-rest-api/src/main/java/org/sonatype/nexus/security/filter/authc/NexusHttpAuthenticationFilter.java [hunk 2]
```diff
@@ -332,16 +332,23 @@ public class NexusHttpAuthenticationFilter
...
-                String username = getNexusConfiguration().getAnonymousUsername();
+                    String username;
 
-                if ( subject != null && subject.isAuthenticated() )
-                {
-                    username = subject.getPrincipal().toString();
-                }
+                    if ( subject != null && subject.isAuthenticated() )
+                    {
+                        username = subject.getPrincipal().toString();
```
Comment (peterlynch) [inline nexus/nexus-rest-api/src/main/java/org/sonatype/nexus/security/filter/authc/NexusHttpAuthenticationFilter.java:343]: might as well protect against NPE here on principal while you are at it. It looks like isAuthenticated would protect against it, but there is no guarantee I would say

File: nexus/nexus-rest-api/src/main/java/org/sonatype/nexus/security/filter/authc/NexusHttpAuthenticationFilter.java [hunk 3]
```diff
@@ -332,16 +332,23 @@ public class NexusHttpAuthenticationFilter
...
+                    }
+                    else
+                    {
+                        username = getNexusConfiguration().getAnonymousUsername();
```
Comment (peterlynch) [inline nexus/nexus-rest-api/src/main/java/org/sonatype/nexus/security/filter/authc/NexusHttpAuthenticationFilter.java:347]: no real guarantee here for no NPE either but that is more the fault of API with no Javadoc or defined behavior.

File: nexus/nexus-rest-api/src/main/java/org/sonatype/nexus/security/filter/authc/NexusHttpAuthenticationFilter.java (hunk 2 above)
Comment (adreghiciu) [inline nexus/nexus-rest-api/src/main/java/org/sonatype/nexus/security/filter/authc/NexusHttpAuthenticationFilter.java:343]: as is just for logging purposes we can do that

File: nexus/nexus-rest-api/src/main/java/org/sonatype/nexus/security/filter/authc/NexusHttpAuthenticationFilter.java (hunk 3 above)
Comment (adreghiciu) [inline nexus/nexus-rest-api/src/main/java/org/sonatype/nexus/security/filter/authc/NexusHttpAuthenticationFilter.java:347]: if nexus config would be null then Nexus should not be running :)

Comment (peterlynch) [inline nexus/nexus-rest-api/src/main/java/org/sonatype/nexus/security/filter/authc/NexusHttpAuthenticationFilter.java:347]: or plexus container didn't wire it correctly...sorry for expecting the worse

**Comment by mcculls** [comment]
+1

**Comment by peterlynch** [comment]
+1 you could protect further against NPE but then what? Throwing RuntimeException of some kind anyways...

**Pull request merged**